integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -2,31 +2,74 @@ use itertools::Itertools;
 use near_async::test_loop::data::TestLoopData;
 use near_async::time::Duration;
 use near_chain_configs::test_genesis::TestGenesisBuilder;
+use near_client::Client;
 use near_o11y::testonly::init_test_logger;
 use near_primitives::epoch_manager::EpochConfigStore;
 use near_primitives::shard_layout::ShardLayout;
+use near_primitives::state_record::StateRecord;
 use near_primitives::types::{AccountId, ShardId};
 use near_primitives::version::{ProtocolFeature, PROTOCOL_VERSION};
-use std::collections::BTreeMap;
+use near_store::ShardUId;
+use std::collections::{BTreeMap, HashMap};
+use std::sync::atomic::{AtomicU64, Ordering};
 use std::sync::Arc;
 
 use crate::test_loop::builder::TestLoopBuilder;
 use crate::test_loop::env::TestLoopEnv;
 use crate::test_loop::utils::ONE_NEAR;
 
-/// Stub for checking Resharding V3.
+fn create_new_shard_layout(base_shard_layout: &ShardLayout, boundary_account: &str) -> ShardLayout {
+    let mut boundary_accounts = base_shard_layout.boundary_accounts().clone();
+    let mut shard_ids: Vec<_> = base_shard_layout.shard_ids().collect();
+    let max_shard_id = *shard_ids.iter().max().unwrap();
+    let last_shard_id = shard_ids.pop().unwrap();
```

Comment:
Sometimes we might want to split a different shard other than the last but it's a story for another day

Reply:
Agreed, let me test non-contiguous shard ids a bit and then we can attempt that.